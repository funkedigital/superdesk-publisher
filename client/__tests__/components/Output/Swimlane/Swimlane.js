import React from "react";
import Swimlane from "../../../../components/Output/Swimlane/Swimlane";
import { render, wait } from "@testing-library/react";
import Store from "../../../../components/Output/Store";
import Publisher from "../../../../__mocks__/publisher";

const publisher = new Publisher();

jest.mock("react-virtualized/styles.css", () => {
  return {};
});
jest.mock("axios");
jest.mock("react-select", () => props => "div");

let api = () => { };

describe("Output/Swimlane/TenantBoard", () => {
  it("renders properly", async () => {
    const { container, getByText } = render(
      <Store.Provider
        value={{
          publisher: publisher,
          api: api,
          tenants: [
            {
              name: 'tenant1', code: 'tenant1', routes: [
                { name: 'testroute1', id: 1 },
                { name: 'testroute2', id: 2 },
              ]
            },
            { name: 'tenant2', code: 'tenant2', routes: [] }
          ],
          filters: { sort: "updated_at", order: "desc" },
          actions: { setFilters: jest.fn(), setArticlesCounts: jest.fn() }
        }}
      >
        <Swimlane />
      </Store.Provider>
    );

    await wait(() =>
      expect(container.querySelector(".sd-loader")).not.toBeInTheDocument(),
    )

    expect(container.firstChild).toMatchSnapshot();
  });
});
