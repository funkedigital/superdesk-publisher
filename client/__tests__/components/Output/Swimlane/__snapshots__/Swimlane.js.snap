// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output/Swimlane/TenantBoard renders properly 1`] = `
<div
  class="sd-kanban-list sd-d-inline-flex sd-flex-grow"
>
  <div
    class="sd-kanban-list__board sd-kanban-list__board--wide"
  >
    <div
      class="sd-kanban-list__board-header"
    >
      <div
        class="sd-list-header"
      >
        <span
          class="sd-list-header__name"
        >
          tenant1
        </span>
        <span
          class="sd-list-header__number badge"
        >
          632
        </span>
      </div>
    </div>
    <div
      class="sd-kanban-list__board-content relative"
      style="display: flex; flex-direction: column;"
    >
      <div
        class="sd-list-item-group sd-shadow--z2"
        style="flex-grow: 1;"
      >
        <div
          style="overflow: visible; height: 0px; width: 0px;"
        >
          <div
            aria-label="grid"
            aria-readonly="true"
            class="ReactVirtualized__Grid ReactVirtualized__List"
            role="grid"
            style="box-sizing: border-box; direction: ltr; height: 0px; position: relative; width: 0px; will-change: transform; overflow-x: hidden; overflow-y: auto;"
            tabindex="0"
          />
        </div>
        <div
          class="resize-triggers"
        >
          <div
            class="expand-trigger"
          >
            <div
              style="width: 1px; height: 1px;"
            />
          </div>
          <div
            class="contract-trigger"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="sd-kanban-list__board sd-kanban-list__board--wide"
  >
    <div
      class="sd-kanban-list__board-header"
    >
      <div
        class="sd-list-header"
      >
        <span
          class="sd-list-header__name"
        >
          tenant2
        </span>
        <span
          class="sd-list-header__number badge"
        >
          632
        </span>
      </div>
    </div>
    <div
      class="sd-kanban-list__board-content relative"
      style="display: flex; flex-direction: column;"
    >
      <div
        class="sd-list-item-group sd-shadow--z2"
        style="flex-grow: 1;"
      >
        <div
          style="overflow: visible; height: 0px; width: 0px;"
        >
          <div
            aria-label="grid"
            aria-readonly="true"
            class="ReactVirtualized__Grid ReactVirtualized__List"
            role="grid"
            style="box-sizing: border-box; direction: ltr; height: 0px; position: relative; width: 0px; will-change: transform; overflow-x: hidden; overflow-y: auto;"
            tabindex="0"
          />
        </div>
        <div
          class="resize-triggers"
        >
          <div
            class="expand-trigger"
          >
            <div
              style="width: 1px; height: 1px;"
            />
          </div>
          <div
            class="contract-trigger"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
